molo/surveys/tests/test_rules.py

```diff
@@ -184,6 +184,38 @@ def test_not_logged_in_user_fails(self):
         self.assertFalse(rule.test_user(self.request))
 
 
+@pytest.mark.django_db
+class TestSurveyNegativeNumberDataRuleSegmentation(
+        TestCase, MoloTestCaseMixin):
+    def setUp(self):
+        # Fabricate a request with a logged-in user
+        # so we can use it to test the segment rule
+        self.mk_main()
+        self.request_factory = RequestFactory()
+        self.request = self.request_factory.get('/')
+        self.request.user = get_user_model().objects.create_user(
+            username='tester', email='tester@example.com', password='tester')
+
+    def test_negative_number_validation(self):
+        # Create survey
+        self.survey = PersonalisableSurvey(title='Test Survey')
+        SurveysIndexPage.objects.first().add_child(instance=self.survey)
+
+        self.neg_number = PersonalisableSurveyFormField.objects.create(
```

Comment:
You're currently only testing the field for a PersonalisableSurvey, you probably need to add a similar test to check this works for a normal survey (MoloSurvey)